src/AdvancedDAC.cpp

```diff
@@ -182,6 +182,15 @@ int AdvancedDAC::stop()
     return 1;
 }
 
+int AdvancedDAC::frequency(uint32_t frequency)
```

Comment:
```suggestion
int AdvancedDAC::frequency(uint32_t const frequency)
```
I suggest using [const](https://isocpp.org/wiki/faq/const-correctness)-correctness as much as possible :wink:

Reply:
Thanks, but I don't think this should be a `const` as I may need to change it in the future (round it up/down) to get the closest possible frequency, also will require changing the HAL config file, and some other code.

Reply:
I disagree. If you need to modify the value in the future, you can assign it to a local copy, then modify that one. Const-correctness is a defensive programming technique prevent foot-shooting of oneself or others. You are (needlessly) rejecting good programming practice now for a future that may or may not come :shrug: .

Reply:
Alex `frequency` is Not passed by reference, modifying it will Not change whatever the user passed, so no foot-shooting ;) `const` here is meaningless, and if even if it's passed by reference I will still need it to be non-const so I can return the actual frequency to the user.

Reply:
If a parameter is passed by reference or not is not the point of importance here :wink: Right now the code snippet is so small that there's little chance of misunderstanding (or one should hope so). If and when the code grows in the future, someone might be assigning a value to what is clearly an input-only parameter.

>The benefit of const correctness is that it prevents you from inadvertently modifying something you didn’t expect would be modified. You end up needing to decorate your code with a few extra keystrokes (the const keyword), with the benefit that you’re telling the compiler and other programmers some additional piece of important semantic information — information that the compiler uses to prevent mistakes and other programmers use as documentation.

Again, this is defensive programming, doing so is caring about potential mistakes in the future, at no cost to yourself (and at no runtime/memory-penalty) except of liberally adding `const` wherever possible.

Reply:
Maybe one little, final appetizer for const-correctness in the end: https://stackoverflow.com/questions/136880/sell-me-const-correctness/136895#136895 :wink:

Reply:
No still not sold on it as it seems overly pedantic to me in this particular case, but not worth arguing over anymore.